Servo's script crate is failing to compile

Not sure how recent this is, but it looks like something we probably need to fix -- I'm somewhat surprised as I thought Servo was configured to compile on stable for Firefox purposes, so changes here are worrisome.

cc @SimonSapin @michaelwoerister @alexcrichton -- anyone know if this is something straightforward to fix? I'd prefer to avoid any large changes to the in-tree copy of Servo...

```
 INFO 2018-05-04T22:02:22Z: collector: failed to benchmark script-servo, recorded: expected success, got exit code: 101

stderr=   Compiling siphasher v0.2.2
   Compiling ref_filter_map v1.0.1
   Compiling num-traits v0.1.40
   Compiling cc v1.0.3
   Compiling lazy_static v0.2.9
   Compiling dtoa v0.4.2
   Compiling cfg-if v0.1.2
   Compiling dom_struct v0.0.1 (file:///tmp/rustc-perf-script-servo.G4HyPmVg5nlD/components/dom_struct)
   Compiling fnv v1.0.5
   Compiling mac v0.1.1
   Compiling script_plugins v0.0.1 (file:///tmp/rustc-perf-script-servo.G4HyPmVg5nlD/components/script_plugins)
error[E0658]: procedural macros cannot be expanded to expressions
  --> components/dom_struct/lib.rs:17:22
   |
17 |       let attributes = quote! {
   |  ______________________^
18 | |         #[derive(DenyPublicFields, DomObject, JSTraceable, MallocSizeOf)]
19 | |         #[must_root]
20 | |         #[repr(C)]
21 | |     };
   | |_____^
   |
   = help: add #![feature(proc_macro_non_items)] to the crate attributes to enable

error: aborting due to previous error

For more information about this error, try `rustc --explain E0658`.
error: Could not compile `dom_struct`.
warning: build failed, waiting for other jobs to finish...
error: build failed

```

--- a/collector/benchmarks/script-servo/components/dom_struct/lib.rs
+++ b/collector/benchmarks/script-servo/components/dom_struct/lib.rs
@@ -2,7 +2,7 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-#![feature(proc_macro)]
+#![feature(proc_macro, proc_macro_non_items)]
 
 extern crate proc_macro;
 

